File: ray-trace-swift/command-line/ray-trace-swift/ray-trace-swift/Ray.swift
//
//  Ray.swift
//  ray-trace-swift
//
//  Created by Luisa Santo on 2/26/19.
//  Copyright © 2019 Luisa Santo. All rights reserved.
//

struct ray {
    var o: vec3_extend
    var d: vec3_extend
    
    init() { self.init(vec3_extend(), vec3_extend()) }
    init(_ o: vec3_extend, _ d: vec3_extend) {
        self.o = o
        self.d = d
    }
    
    var origin: vec3_extend { return o }
    var direction: vec3_extend { return d }
    func pointAtParameter(_ t: Float) -> vec3_extend {
        return o + t*d
    }
    
}

File: Assignment1/Assets/Ray.cpp
#include "Ray.h"

Ray::Ray(Point origin, Point direction) {
	_origin = origin;
	//Normalize vector
	direction.normalize();
	_direction = direction;
}

Point Ray::pointAtParameter(float t) {

	// o + d * t
	Point origin = getOrigin();
	Point direction = getDirection();
	return origin.add(direction.multiply(t));
}